### PR TITLE
gemmini-cisc ISA spike/software done. some unrelated bareMetalC are f…

### PR DESCRIPTION
**Pull request by ssteffl**
added gemmini-cisc ISA. passing functional simulations

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -2,6 +2,7 @@ build/
 *.gch
 autom4te.cache/
 .*.swp
+*.swo
```
Comment (hngenc) [inline .gitignore:5]: What is an swo file? Is this related to the CISC instruction, or is this new?

Comment (ssteffl) [reply .gitignore:5]: vim alternate swap file, when the *.swp file already exists

**Comment by ssteffl** [comment]
> I made one minor comment, but it looks good to me overall.
> 
> However, can we leave this in a new branch (i.e. `cisc`) until the hardware is ready? I prefer to only merge changes into `esp-sim-isa/master` when the corresponding hardware is in `gemmini/dev`

sure, sounds fine to me
